src/object/reply.c

```diff
@@ -64,27 +63,26 @@ void reply_pop(reply_t *reply, tcb_t *tcb)
     word_t next_ptr = call_stack_get_callStackPtr(reply->replyNext);
     word_t prev_ptr = call_stack_get_callStackPtr(reply->replyPrev);
 
-    if (likely(next_ptr != 0)) {
-        assert(call_stack_get_isHead(reply->replyNext));
+    assert(call_stack_get_isHead(reply->replyNext));
```

Comment:
We should drop this `assert` as well, it is explicitly checked in the `if` just before `reply_pop` is called.